lib/polygeist/Passes/ParallelLoopDistribute.cpp

```diff
@@ -932,24 +903,26 @@ struct InterchangeForIfPFor : public OpRewritePattern<ParallelOpType> {
       return failure();
     }
 
-    if (!arePreceedingOpsRecomputable(&*op, lastOp)) {
+    if (!arePreceedingOpsRecomputable(lastOp)) {
       LLVM_DEBUG(DBGS() << "[interchange] found a nonrecomputable op\n");
       return failure();
     }
 
     // In the GPU model, the trip count of the inner sequential containing a
     // barrier must be the same for all threads. So read the value written by
     // the first thread outside of the loop to enable interchange.
+
+    // Replicate the recomputable ops in case the condition or bound of lastOp
+    // is getting "recomputed" TODO dead code elimination should get rid of
+    // unneeded ops here, but is it better to do the check if it is needed
+    // ourselves?
```

Comment:
DCE should be able to get rid of automatically during the pattern match as a fold, so don't feel the need to add.